WalletWasabi/Crypto/ZeroKnowledge/RandomizedCommitments.cs

```diff
@@ -5,27 +5,33 @@ namespace WalletWasabi.Crypto.ZeroKnowledge
 {
 	public class RandomizedCommitments
 	{
-		public static RandomizedCommitments RandomizeMAC(MAC mac, Scalar z, GroupElement Ma)
-		{
-			GroupElement Randomize(GroupElement G, GroupElement M) => M + z * G;
-			return new RandomizedCommitments(
-				Ca:  Randomize(Generators.Ga, Ma),
-				Cx0: Randomize(Generators.Gx0, mac.U),
-				Cx1: Randomize(Generators.Gx1, mac.T * mac.U),
-				CV:  Randomize(Generators.GV, mac.V));
-		}
-
-		private RandomizedCommitments(GroupElement Ca, GroupElement Cx0, GroupElement Cx1, GroupElement CV)
+		private RandomizedCommitments(GroupElement Ca, GroupElement Cx0, GroupElement Cx1, GroupElement CV, GroupElement S)
```

Comment:
since `ComputeZ` is called by the issuer, who only has access to the group elements, this constructor should be public

Reply:
maybe Z can just be a property that is computed in the constructor/static helpers, with one path, `RandomizeMAC` requiring (iparams, mac, amount, randomness) as args, and the other requiring (Ca, Cx0, Cx1, CV, S, coordinator secret key)?